### PR TITLE
Made sure toast does not add it's container until the consumer wants to display the toast.

### PR DESCRIPTION
**Pull request by haavamoa**
## Added / Fixed

- Removed `Toast.Initialize`
- Removed `Appearing` subscription to the current page.
- Kept `Disappearing` subscription on the current page in order to dismiss the toast when the current page is navigated out of.
- Cleaned up some unnecessary null-checks.
- Removed `MainThread.Invoke` as this did not have any effect when testing on Android and iOS.
- Removed event subscription to use the `ToastAction` when a toast is tapped.

## Todo List

- [X] I have tested on an Android device.
- [X] I have tested on an iOS device.
- [ ] I have added unit tests

>*Please add pictures / Gifs if possible*
>*Todo List can be checked by putting a `X` inside the brackets*
>*Remember to update our `wiki` after this PR is merged*

### CODE REVIEW

File: src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs [hunk 1]
```diff
@@ -109,20 +84,28 @@ private async Task HideToast(ContentPage currentPage, bool removeContainer)
...
-            return toastContainer;
-        }
+            m_currentPageWithToast.Disappearing += OnPageDissapearing;
 
-        private static ContentPage? GetCurrentContentPage()
+            return toastContainer;
+        }
+
+        private void OnPageDissapearing(object sender, EventArgs e)
+        {
```
Comment (rizanzaky) [inline src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs:103]: Typo

File: src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs [hunk 2]
```diff
@@ -178,12 +161,10 @@ private ToastView GetToast(string text, ToastOptions options, ToastLayout layout
...
 
-            var tapGesture = new TapGestureRecognizer();
-            tapGesture.Tapped += (s, e) =>
-            {
-                toast.ToastOptions.ToastAction?.Invoke();
-            };
-            toast.GestureRecognizers.Add(tapGesture);
+            if(toast.ToastOptions.ToastAction != null)
+            {
+                toast.GestureRecognizers.Add(new TapGestureRecognizer() { Command = new Command((toast.ToastOptions.ToastAction)) });
```
Comment (rizanzaky) [inline src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs:166]: Now `GetToast` can be made `static`

File: src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs [hunk 3]
```diff
@@ -109,20 +84,28 @@ private async Task HideToast(ContentPage currentPage, bool removeContainer)
...
                 toastContainer = new Grid();
-                RegisterName(currentPage.Id.ToString(), toastContainer);
+                RegisterName(m_currentPageWithToast.Id.ToString(), toastContainer);
 
                 // old content
-                var oldContent = currentPage.Content;
+                var oldContent = m_currentPageWithToast.Content;
 
                 // set new content
-                await MainThread.InvokeOnMainThreadAsync(() => { currentPage.Content = toastContainer; });
```
Comment (rizanzaky) [inline src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs:118]: Yes, invoking on `MainThread` can be removed and let the user wrap it if it's needed, just like in other `Xamarin.Forms` controls. (Needed in cases where the Toast is displayed from a ViewModel after a Background service thread invokes it)

Comment (rizanzaky) [inline src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs:118]: Yes, invoking on `MainThread` can be removed and let the user wrap it if it's needed, just like in other `Xamarin.Forms` controls. (Needed in cases where the Toast is displayed from a ViewModel after a Background service thread invokes it)

File: src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs [hunk 4]
```diff
@@ -178,12 +161,10 @@ private ToastView GetToast(string text, ToastOptions options, ToastLayout layout
                 Margin = new Thickness(layout.HorizontalMargin, layout.PositionY, layout.HorizontalMargin, 0)
             };
 
-            var tapGesture = new TapGestureRecognizer();
-            tapGesture.Tapped += (s, e) =>
-            {
-                toast.ToastOptions.ToastAction?.Invoke();
-            };
-            toast.GestureRecognizers.Add(tapGesture);
+            if(toast.ToastOptions.ToastAction != null)
```
[the remark this answers is not in the document]
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs:164]: Be more specific. What do you mean?

[the remark this answers is not in the document]
Comment (haavamoa) [reply src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs]: Do you want a `space` before the condition?

[the remark this answers is not in the document]
Comment (rizanzaky) [reply src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs]: I see `if(` has no space between, so I was wondering if code cleanup is followed, I use ReSharper, so not sure

**Pull request merged**